File: 第1章 古算趣题/1.5 李白沽酒/李白沽酒.py
'''
程序：李白沽酒
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
def main():
    '''李白沽酒'''
    n = 0                      # 酒量
    for i in range(4):
        n = (n + 6) / 2        # 反推瓶内酒量
    print('瓶内原有酒%s升' % n)

if __name__ == '__main__':
    main()

File: 第1章 古算趣题/1.5 李白沽酒/李白沽酒.cpp
/*
程序：李白沽酒 
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
#include <bits/stdc++.h>
using namespace std;
// 李白沽酒 
int main()
{
	float n = 0;                  // 酒量 
	for (int i = 0; i < 4; i++) { // 反推瓶内酒量 
		n = (n + 6) / 2;
	}
	cout << "瓶内原有酒" << n << "升" << endl;
    return 0;
}
